### ISSUE TITLE
[FTBFS into debian] A test randomly fails.

### ISSUE THREAD

**Issue by P-EB** [aio-libs/aiosmtpd#133]
Dear developers,

Sometimes, the testing part of aiosmtpd fails, because of a specific test.
Here is, in that case, the error that occurs.

```
======================================================================
ERROR: test_debug_3 (aiosmtpd.tests.test_main.TestMain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-aiosmtpd-1.1/.pybuild/pythonX.Y_3.6/build/aiosmtpd/tests/test_main.py", line 140, in test_debug_3
    main(('-n', '-ddd'))
  File "/build/python-aiosmtpd-1.1/.pybuild/pythonX.Y_3.6/build/aiosmtpd/main.py", line 140, in main
    loop.create_server(factory, host=args.host, port=args.port))
  File "/usr/lib/python3.6/asyncio/base_events.py", line 465, in run_until_complete
    raise RuntimeError('Event loop stopped before Future completed.')
RuntimeError: Event loop stopped before Future completed.

----------------------------------------------------------------------
 ```
This error does prevent aiosmtpd to build from source into debian, and so prevents the reproductibility tests to succeed.

Can you have a look at it please?

**Comment by warsaw** [comment]
We just saw this on a recent PR, but with a different test in the same test class.  So it's a real problem.

**Comment by P-EB** [comment]
Would you require some assistance in order to solve this issue?

I'm not used to the code base, but I can dive into it!

**Comment by warsaw** [comment]
@P-EB Yes!  I haven't been very active on this project lately (mostly because of free time and because what we have solves my needs), but I would be very happy to review fixes for this problem.

**Comment by waynew** [comment]
@P-EB have you looked into this at all? Obviously it's been a while :) Do you have any interest in continuing, or should we consider this issue up for grabs?

**Comment by P-EB** [comment]
@waynew I did investigate a lot. But as my PhD goes, I didn't find the time recently to try again.

Any help is welcome, and the issue is still occurring.

**Comment by P-EB** [comment]
@waynew did you find some time to think about this? I'm still out of ideas and I'm not able to reproduce the issue on a deterministic base (which kind of ruins all my attempts to investigate).

**Comment by waynew** [comment]
@P-EB I haven't looked into this at all. I've got some time right now though, so I'll poke at it with a stick

**Comment by waynew** [comment]
Hah. That didn't take long. And helpfully enough, we already have a comment that calls this out:

https://github.com/aio-libs/aiosmtpd/blob/master/aiosmtpd/tests/test_main.py#L47-L52

If you decrease the sleep time to like `0.01` it should error consistently, at least it appears to for me. I'm not sure what the *right* way to fix that is - it seemed like `0.5` was enough of a timeout bump, but it feels like there should be a better way to do what we're trying to do :suspect: 

@P-EB I'm not sure how familiar you are with asyncio, but if someone is and knows a better way to accomplish what we're trying to do there, that would be 👍

**Comment by P-EB** [comment]
@warsaw do you remember about this? On a first glance it seems that @waynew is right. I wonder how I missed this comment while I was reviewing the code in search for this bug. Probably I was tired at that point.

**Comment by warsaw** [comment]
Yeah, unfortunately I didn't really know of a better way back then and don't now. ;)

**Comment by P-EB** [comment]
That's fine, let's try with increasing this delay! :)

**Comment by P-EB** [comment]
Ok, for me it worked fine with 0.5 instead of 0.1.

**Comment by pepoluan** [comment]
This is implemented in 602e6fd , also part of PR #192 

I increased the delay to `1.0`, though. Just in case.

**Comment by pepoluan** [comment]
Fixed in #195 (which was cherry-picked from #192) which was already merged, so I'm closing this issue.

**Comment by P-EB** [comment]
Thanks!
